lib/textParsers.js

```diff
@@ -1,49 +1,68 @@
 var arrayParser = require(__dirname + "/arrayParser.js");
 
+var fullDateMatcher = /^(\d{1,})-(\d{2})-(\d{2}) (\d{2}):(\d{2}):(\d{2})(\.\d{1,})?([Z+\-]([^\\w]*))?/;
+var dateOnlyMatcher = /^(\d{1,})-(\d{2})-(\d{2})/;
+
 //parses PostgreSQL server formatted date strings into javascript date objects
 var parseDate = function(isoDate) {
-  //TODO this could do w/ a refactor
-  var dateMatcher = /(\d{1,})-(\d{2})-(\d{2}) (\d{2}):(\d{2}):(\d{2})(\.\d{1,})?/;
-
-  var match = dateMatcher.exec(isoDate);
-  //could not parse date
-  if(!match) {
-    dateMatcher = /^(\d{1,})-(\d{2})-(\d{2})$/;
-    match = dateMatcher.test(isoDate);
-    if(!match) {
+
+  var isFullDate = true;
+  var match = fullDateMatcher.exec(isoDate);
+  if (match === null) {
+    match = dateOnlyMatcher.exec(isoDate);
+    if (match === null) {
       return null;
-    } else {
-      //it is a date in YYYY-MM-DD format
-      //add time portion to force js to parse as local time
-      return new Date(isoDate + ' 00:00:00');
     }
+    isFullDate = false;
   }
+
+  var date;
   var isBC = /BC$/.test(isoDate);
   var _year = parseInt(match[1], 10);
   var isFirstCentury = (_year > 0) && (_year < 100);
   var year = (isBC ? "-" : "") + match[1];
-
   var month = parseInt(match[2],10)-1;
   var day = match[3];
-  var hour = parseInt(match[4],10);
-  var min = parseInt(match[5],10);
-  var seconds = parseInt(match[6], 10);
-
-  var miliString = match[7];
-  var mili = 0;
-  if(miliString) {
-    mili = 1000 * parseFloat(miliString);
+
+  if (isFullDate) {
+    var hour = parseInt(match[4],10);
+    var min = parseInt(match[5],10);
+    var seconds = parseInt(match[6], 10);
+
+    var miliString = match[7];
+    var mili = 0;
+    if (miliString) {
+      mili = 1000 * parseFloat(miliString);
+    }
+
+    if (match[8]) {
+      var tzAdjust = parseTimeZone(match[8]);
+      var utcOffset = Date.UTC(year, month, day, hour, min, seconds, mili);
+      date = new Date(utcOffset - (tzAdjust * 1000));
+    } else {
+      date = new Date(year, month, day, hour, min, seconds, mili);
+    }
+  } else {
+    date = new Date(year, month, day);
```

Comment:
This should read
```
date = new Date(Date.UTC(year, month, day));
```
as the raw postgres `date` datatype should always be interpreted as UTC (*not* the local timezone as `new Date` would).